src/main/scala/ai/privado/languageEngine/ruby/processor/RubyProcessor.scala

```diff
@@ -90,199 +92,130 @@ class RubyProcessor(
+            try {
+              finalResult += Await.result(result, privadoInput.rubyParserTimeout.seconds)
+            } catch {
+              case ex: TimeoutException =>
+                statsRecorder.justLogMessage(s"Parser timed out for file -> '${task.file}'")
+                timeoutFileCount += 1
+              case ex: Exception =>
+                statsRecorder.justLogMessage(s"Error while parsing file -> '${task.file}'")
+                errorFileCount += 1
             }
           }
-
-          JSONExporter.fileExport(
-            cpg,
-            outputFileName,
-            sourceRepoLocation,
-            dataflowMap,
-            ruleCache,
-            taggerCache,
-            dataFlowCache.getDataflowAfterDedup,
-            privadoInput,
-            monolithPrivadoJsonPaths = monolithPrivadoJsonPaths,
-            s3DatabaseDetailsCache,
-            appCache,
-            propertyFilterCache,
-            databaseDetailsCache
-          ) match {
-            case Left(err) =>
-              MetricHandler.otherErrorsOrWarnings.addOne(err)
-              errorMsg += err
-            case Right(_) =>
-              println(s"Successfully exported output to '${appCache.localScanPath}/$outputDirectoryName' folder")
-              logger.debug(
-                s"Total Sinks identified : ${cpg.tag.where(_.nameExact(Constants.catLevelOne).valueExact(CatLevelOne.SINKS.name)).call.tag.nameExact(Constants.id).value.toSet}"
-              )
-          }
-
-          // Check if any of the export failed
-          if (errorMsg.toList.isEmpty)
-            Right(())
-          else
-            Left(errorMsg.toList.mkString("\n"))
-
-        } catch {
-          case ex: Exception =>
-            logger.error("Error while processing the CPG after source code parsing ", ex)
-            MetricHandler.setScanStatus(false)
-            Left("Error while parsing the source code: " + ex.toString)
-        } finally {
-          cpg.close()
-          import java.io.File
-          val cpgFile = new File(cpgconfig.outputPath)
-          println(s"\n\n\nBinary file size -- ${cpgFile.length()} in Bytes - ${cpgFile.length() * 0.000001} MB\n\n\n")
+          statsRecorder.justLogMessage(s"No of files skipped because of timeout - '$timeoutFileCount'")
+          statsRecorder.justLogMessage(s"No of files that are skipped because of error - '$errorFileCount'")
+          finalResult.toList
         }
 
-      case Failure(exception) =>
-        logger.error("Error while parsing the source code!", exception)
-        logger.debug("Error : ", exception)
-        MetricHandler.setScanStatus(false)
-        Left("Error while parsing the source code: " + exception.toString)
+        new io.joern.rubysrc2cpg.deprecated.ParseInternalStructures(parsedFiles, cpg.metaData.root.headOption)
+          .populatePackageTable()
+        val astCreationPass =
+          new AstCreationPass(cpg, parsedFiles, RubySrc2Cpg.packageTableInfo, config)
```

Comment:
I see the shared nature of `packageTableInfo` can cause issue on privado-core unit tests. Ignore for now

Reply:
Yes this can be shifted into cache.